### PR TITLE
refactor: chaos caller validation

### PR DESCRIPTION
**Pull request by alanshaw**
This PR refactors the `CallerValidation` method in the chaos actor allowing it to accept caller addresses and caller type CIDs. It also adds a new branch that allows testing caller validation for internal sends to self and additionally that `Runtime.Message().Receiver()` returns ID addresses not robust addresses.

This allows us to create test vectors for https://github.com/filecoin-project/test-vectors/issues/128 and also allows us to add a few vectors that test caller validation for non-nil addresses and type cids.

### CODE REVIEW

File: conformance/chaos/actor.go [hunk 1]
```diff
@@ -31,10 +29,16 @@ type Actor struct{}
...
+	// CallerValidationBranchNone causes no caller validation to take place.
 	CallerValidationBranchNone CallerValidationBranch = iota
+	// CallerValidationBranchTwice causes Runtime.ValidateImmediateCallerAcceptAny to be called twice.
 	CallerValidationBranchTwice
-	CallerValidationBranchAddrNilSet
-	CallerValidationBranchTypeNilSet
+	// CallerValidationBranchIs causes caller validation against CallerValidationArgs.Addrs.
+	CallerValidationBranchIs
+	// CallerValidationBranchIsReceiver causes validation that the caller was also the receiver.
+	CallerValidationBranchIsReceiver
```
Comment (raulk) [inline conformance/chaos/actor.go:39]:
```suggestion
	CallerValidationBranchIsSelf
```

I think it would be more deterministic if we used `CallerValidationBranchIs` with the explicit address we expect? We do seem to always know it ahead of time.
